rust_snuba/src/processors/outcomes.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashSet;
+
+use crate::types::{BadMessage, BytesInsertBatch, KafkaMessageMetadata};
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::Deserialize;
+
+pub fn process_message(
+    payload: KafkaPayload,
+    _metadata: KafkaMessageMetadata,
+) -> Result<BytesInsertBatch, BadMessage> {
+    if let Some(payload_bytes) = payload.payload {
+        let msg: FromOutcomeMessage = serde_json::from_slice(&payload_bytes).map_err(|err| {
+            log::error!("Failed to deserialize message: {}", err);
+            BadMessage
+        })?;
+
+        return Ok(BytesInsertBatch { rows: vec![] });
+    }
+
+    Err(BadMessage)
+}
+
+#[derive(Debug, Default, Deserialize)]
+struct FromOutcomeMessage {
+    timestamp: String,
+
+    org_id: Option<u64>,
+    project_id: Option<u64>,
+    key_id: Option<u64>,
+    outcome: u8,
+
+    reason: Option<String>,
+    event_id: Option<String>,
+    category: Option<u8>,
+    quantity: Option<u64>,
+}
+
+struct Outcome {
+    timestamp: u64,
+
+    org_id: u64,
+    project_id: u64,
+    key_id: Option<u64>,
+    outcome: u8,
+
+    reason: Option<String>,
+    event_id: String,
+    category: u8,
+    quantity: u64,
+}
+
+impl TryFrom<FromOutcomeMessage> for Outcome {
+    type Error = BadMessage;
+
+    fn try_from(from: FromOutcomeMessage) -> Result<Self, Self::Error> {
+        const OUTCOME_ABUSE: u8 = 4;
+        const OUTCOME_CLIENT_DISCARD: u8 = 5;
+        let client_discard_reasons: HashSet<&str> = vec![
+            "queue_overflow",
+            "cache_overflow",
+            "ratelimit_backoff",
+            "network_error",
+            "before_send",
+            "event_processor",
+            "sample_rate",
+            "send_error",
+            "internal_sdk_error",
+            "insufficient_data",
+            "backpressure",
+        ]
+        .into_iter()
+        .collect();
```

Comment:
This set doesn't need to be created on every call of this function. I think this could just be a constant array; at 11 elements just searching through it should be fine.